Legacy/bonej/src/main/java/org/bonej/plugins/UsageReporter.java

```diff
@@ -157,8 +157,15 @@ public void send() {
 				utmhid + utmr + utmp + utmac + utmcc);
 			final URLConnection uc = url.openConnection();
 			uc.setRequestProperty("User-Agent", userAgentString());
-			if (IJ.debugMode) {
-				logRequest(url, uc);
+			try (final BufferedReader reader = new BufferedReader(new InputStreamReader(
+				uc.getInputStream()))) {
+					if (IJ.debugMode) {
+						IJ.log(url.toString());
+						IJ.log(uc.getRequestProperty("User-Agent"));
+						reader.lines().forEach(IJ::log);
+					}
+			} catch (final IOException e) {
+				IJ.error(e.getMessage());
 			}
```

Comment:
I am probably being silly but I don't understand how the changes make a difference.

The only thing I see changing when `IJ.debugMode` is `false` is that the `reader` variable gets constructed, and then goes out of scope at the end of the `send()` function?

Reply:
Right - no idea either. I knocked out some lines to see what is essential and it's this:
```
final BufferedReader reader = new BufferedReader(new InputStreamReader(uc.getInputStream()))
```
Which I interpret to mean that the request isn't actually sent until the `BufferedReader` tries to read from the `URLConnection`.